packages/fxa-content-server/tests/intern.js

```diff
@@ -0,0 +1,181 @@
+// Sometimes it means real "stage" or real production configuration, but
+// sometimes it also means fxa-dev style boxes like "latest". Configuration
+// parameter "fxaDevBox" can be used as a crude way to distinguish between
+// two.
+const fxaProduction = !!args.fxaProduction;
+const fxaDevBox = !!args.fxaDevBox;
+
+const fxaToken = args.fxaToken || 'http://';
+const asyncTimeout = parseInt(args.asyncTimeout || 10000, 10);
+
+// On Circle, we bail after the first failure.
+// args.bailAfterFirstFailure comes in as a string.
+const bailAfterFirstFailure = args.bailAfterFirstFailure === 'true';
+
+const testProductId = args.testProductId || 'prod_GqM9ToKK62qjkK';
+const testPlanId = args.testPlanId || 'plan_GqM9N6qyhvxaVk';
+
+// Intern specific options are here: https://theintern.io/docs.html#Intern/4/docs/docs%2Fconfiguration.md/properties
+const config = {
+  asyncTimeout: asyncTimeout,
+  bail: bailAfterFirstFailure,
+  defaultTimeout: 45000, // 30 seconds just isn't long enough for some tests.
+  environments: {
+    browserName: 'firefox',
+    fixSessionCapabilities: 'no-detect',
+    usesHandleParameter: true,
+  },
+  filterErrorStack: true,
+
+  fxaAuthRoot: fxaAuthRoot,
+  fxaContentRoot: fxaContentRoot,
+  fxaSettingsV2Root: fxaSettingsV2Root,
+  fxaDevBox: fxaDevBox,
+  fxaEmailRoot: fxaEmailRoot,
+  fxaOAuthApp: fxaOAuthApp,
+  fxaOAuthRoot: fxaOAuthRoot,
+  fxaProduction: fxaProduction,
+  fxaProfileRoot: fxaProfileRoot,
+  fxaToken: fxaToken,
+  fxaTokenRoot: fxaTokenRoot,
+  fxaUntrustedOauthApp: fxaUntrustedOauthApp,
+  fxaPaymentsRoot,
+
+  pageLoadTimeout: 30000,
+  reporters: [
+    {
+      name: 'junit',
+      options: {
+        filename: output,
+      },
+    },
+    'runner',
+  ],
+  serverPort: 9091,
+  serverUrl: 'http://localhost:9091',
+  socketPort: 9077,
+  tunnelOptions: {
+    drivers: [
+      {
+        name: 'firefox',
+      },
+    ],
+  },
+
+  testProductId,
+  testPlanId,
+};
+
+if (args.grep) {
+  config.grep = new RegExp(args.grep, 'i');
+}
+
+if (args.suites) {
+  switch (args.suites) {
+    case 'server':
+    case 'server-resources':
+      config.functionalSuites = [];
+      config.node = {
+        suites: testsServer,
+      };
+      config.tunnelOptions = {};
+      config.environments = {
+        browserName: 'node',
+      };
+      config.reporters = 'pretty';
+      if (args.suites === 'server-resources') {
+        config.node.suites = testsServerResources;
+      }
+      break;
+  }
+}
+
+if (args.unit) {
+  config.functionalSuites.unshift('tests/functional/mocha.js');
+}
+
+if (args.groups !== undefined && args.groupIndex !== undefined) {
+  const index = args.groupIndex;
+  const groups = args.groups;
+  const groupSize = Math.ceil(config.functionalSuites.length / groups);
+  const originalSuiteSize = config.functionalSuites.length;
+  const targeted = config.functionalSuites.splice(groupSize * index, groupSize);
```

Comment:
Not sure if it matters, but looks like `config.functionalSuites` is set to an empty array on L104 above. So not sure if this will always mean... `groupSize === 0`, `originalSuiteSize === 0`, and `targeted === []` (although, not sure about the last one just by eyeballing the code).